src/osvr/Util/Logger.cpp

```diff
@@ -131,95 +139,90 @@ namespace util {
-                           : detail::LineLogger{name_, LogLevel::trace, msg};
+        Logger::StreamProxy Logger::trace(const char *msg) {
+            StreamProxy proxy(*this, LogLevel::trace);
+            proxy << msg;
+            return proxy;
         }
 
-        detail::LineLogger Logger::debug(const char *msg) {
-            return logger_ ? detail::LineLogger{logger_->debug(msg)}
-                           : detail::LineLogger{name_, LogLevel::debug, msg};
+        Logger::StreamProxy Logger::debug(const char *msg) {
+            StreamProxy proxy(*this, LogLevel::trace);
+            proxy << msg;
+            return proxy;
         }
 
-        detail::LineLogger Logger::info(const char *msg) {
-            return logger_ ? detail::LineLogger{logger_->info(msg)}
-                           : detail::LineLogger{name_, LogLevel::info, msg};
+        Logger::StreamProxy Logger::info(const char *msg) {
+            StreamProxy proxy(*this, LogLevel::trace);
+            proxy << msg;
+            return proxy;
         }
 
-        detail::LineLogger Logger::notice(const char *msg) {
-            return logger_ ? detail::LineLogger{logger_->notice(msg)}
-                           : detail::LineLogger{name_, LogLevel::notice, msg};
+        Logger::StreamProxy Logger::notice(const char *msg) {
+            StreamProxy proxy(*this, LogLevel::trace);
+            proxy << msg;
+            return proxy;
         }
 
-        detail::LineLogger Logger::warn(const char *msg) {
-            return logger_ ? detail::LineLogger{logger_->warn(msg)}
-                           : detail::LineLogger{name_, LogLevel::warn, msg};
+        Logger::StreamProxy Logger::warn(const char *msg) {
+            StreamProxy proxy(*this, LogLevel::trace);
+            proxy << msg;
+            return proxy;
         }
 
-        detail::LineLogger Logger::error(const char *msg) {
-            return logger_ ? detail::LineLogger{logger_->error(msg)}
-                           : detail::LineLogger{name_, LogLevel::error, msg};
+        Logger::StreamProxy Logger::error(const char *msg) {
+            StreamProxy proxy(*this, LogLevel::trace);
+            proxy << msg;
+            return proxy;
         }
 
-        detail::LineLogger Logger::critical(const char *msg) {
-            return logger_ ? detail::LineLogger{logger_->critical(msg)}
-                           : detail::LineLogger{name_, LogLevel::critical, msg};
+        Logger::StreamProxy Logger::critical(const char *msg) {
+            StreamProxy proxy(*this, LogLevel::trace);
+            proxy << msg;
+            return proxy;
         }
 
         // logger.info() << ".." call  style
-        detail::LineLogger Logger::trace() {
-            return logger_ ? detail::LineLogger{logger_->trace()}
-                           : detail::LineLogger{name_, LogLevel::trace};
+        Logger::StreamProxy Logger::trace() {
+            return { *this, LogLevel::trace };
```

Comment:
Is this somehow an implicit conversion? Does this save us a move-construction? The syntax looks weird, and it's not just because you didn't run clang-format here :-P